### DISCUSSION TITLE
OSPP 2024 Project Proposal: Add UI for Kvrocks Controller

### DISCUSSION THREAD

**Discussion by jihuayu** [apache/kvrocks-controller#166] [General]
Related issues https://github.com/apache/kvrocks-controller/issues/135

We are preparing to submit our project proposal for the [OSPP](https://summer-ospp.ac.cn/). Below is the project description, and we request your feedback and suggestions.

You can find other projects of OSPP at [this link](https://summer-ospp.ac.cn/org/projectlist?lang=zh&pageNum=1&pageSize=50).

----

### Project Name:
Add Web Management UI to Apache Kvrocks Controller

### Project Description:
Add a web management UI to Apache Kvrocks Controller, allowing users to manage clusters through a web page.

### Project Notes:
Apache Kvrocks is a distributed key value NoSQL database that uses RocksDB as storage engine and is compatible with Redis protocol.
Apache Kvrocks Controller is a cluster management tool for Apache Kvrocks.

### Project Deliverables:

- https://github.com/apache/kvrocks-controller/issues/138
- https://github.com/apache/kvrocks-controller/issues/139
- https://github.com/apache/kvrocks-controller/issues/140
- https://github.com/apache/kvrocks-controller/issues/142
- https://github.com/apache/kvrocks-controller/issues/141

### Project Technical Requirements:
1. Proficiency in React & TypeScript, familiarity with web frontend development
2. Understanding of Kvrocks or Redis cluster concepts

### Project Repository:
https://github.com/apache/kvrocks-controller

**Comment by CulturalProfessor** [comment]
Hey guys I wanted to apply for this project and was working on a draft [Proposal](https://docs.google.com/document/d/1BldEAh8nkEN6nt63tgMNH-r1AqMkxsnxlk2Plo49Pok/edit?usp=sharing) and [Design](https://www.figma.com/design/7yCVi4SPHqW8tSHHzwILXy/Kvrocks-Controller-Wireframe?t=hO2JyguXAsZJHEae-0), can you please provide some feedback and suggestions ?

**Reply by jihuayu** [reply]
@git-hulk Can you take a look?

**Reply by git-hulk** [reply]
Sure, but the current design only contains the cluster creation part, it will be better if it can include the shard and slot/node management.

**Reply by CulturalProfessor** [reply]
> Sure, but the current design only contains the cluster creation part, it will be better if it can include the shard and slot/node management.

Yeah, working on it, will update soon.

**Reply by git-hulk** [reply]
@CulturalProfessor Thanks for your great work.

**Reply by CulturalProfessor** [reply]
> @CulturalProfessor Thanks for your great work.

Guys I designed a basic UI for shard and node similar to cluster one, please suggest any modifications ?

**Reply by jihuayu** [reply]
@CulturalProfessor Looks like good. Don't forget 2 steps confirmation when delete node/cluster/namespace.

**Reply by CulturalProfessor** [reply]
> @CulturalProfessor Looks like good. Don't forget 2 steps confirmation when delete node/cluster/namespace.

Ok sure, will update the design
